cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -355,6 +384,27 @@ func (m *asgCache) regenerate() error {
 	return nil
 }
 
+func (m *asgCache) createPlaceholdersForDesiredNonStartedInstances(groups []*autoscaling.Group) []*autoscaling.Group {
+	for _, g := range groups {
+		desired := *g.DesiredCapacity
+		real := int64(len(g.Instances))
+		if desired <= real {
+			continue
+		}
+
+		for i := real; i < desired; i++ {
+			id := fmt.Sprintf("%s%d", placeholderInstanceNamePrefix, i)
```

Comment:
If you have multiple ASGs with placeholders you'll duplicate ids. I'm not sure if it will create problems or not, but it's probably worth testing.

Reply:
Good point, **I'm not 100% sure it'll hold true for this PR as we're running a patched version of CA v1.3.8** but we've had this occur over the past couple of days. 

Behaviour we've seen: The CA only picks up that there's one unregistered node called `i-placeholder-0` at a time despite 2 ASGs having placeholder nodes with the name `i-placeholder-0` so only attempts to decrement the size of one of them, but not the other, until the first group is resolved.

@piontec how about adding the `AutoScalingGroupName` to the end of the placeholder instance name to distinguish these from each other?

Reply:
Yes, @MaciekPytel has an excellent point. Seems like this is an easy fix (as Guy suggests, simply add the autoscaling group name to the placeholder instance name template).

Reply:
Yes, great catch, fixing